core/http-auth-aws-event-stream/src/main/java/software/amazon/awssdk/http/auth/aws/eventstream/internal/SigV4DataFramePublisher.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.eventstream.internal;
+
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.computeSignature;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.deriveSigningKey;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.hash;
+
+import java.nio.ByteBuffer;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Function;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.auth.aws.signer.CredentialScope;
+import software.amazon.awssdk.http.auth.aws.util.SignerConstant;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.utils.BinaryUtils;
+import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.internal.MappingSubscriber;
+import software.amazon.eventstream.HeaderValue;
+import software.amazon.eventstream.Message;
+
+/**
+ * A publisher which adapts a publisher by SigV4-signing each frame, and appends an empty trailing frame.
+ */
+@SdkInternalApi
+public final class SigV4DataFramePublisher implements Publisher<ByteBuffer> {
+
+    private static final Logger LOG = Logger.loggerFor(SigV4DataFramePublisher.class);
+    private static final String CHUNK_SIGNATURE = ":chunk-signature";
+    private static final int PAYLOAD_TRUNCATE_LENGTH = 32;
+
+    private final Publisher<ByteBuffer> sigv4Publisher;
+
+    public SigV4DataFramePublisher(Publisher<ByteBuffer> publisher,
+                                   AwsCredentialsIdentity credentials,
+                                   CredentialScope credentialScope,
+                                   String signature,
+                                   Clock signingClock) {
+
+        // Adapt the publisher with a trailing-empty frame publisher
```

Comment:
This comment is not needed, same as line 64. We should minimize the use of comments by making the code self-explanatory.

Reply:
NVM, seems like this is copied from existing file.